### PR TITLE
#trivial MOBILE-966 DateField refactor

### PR DESCRIPTION
**Pull request by mclomello**
@victorg1991 @dgarciasarai

### CODE REVIEW

File: android/library/src/main/java/com/liferay/mobile/screens/ddl/model/DateField.java [hunk 1]
```diff
@@ -66,29 +65,17 @@ protected Date convertFromString(String stringValue) {
...
-            }
-        } catch (ParseException e) {
-            LiferayLogger.e("Error parsing date " + stringValue);
+        int lastSeparator = stringValue.lastIndexOf('/');
+
+        if (stringValue.contains("-")) {
+            return convertDate("yyyy-MM-dd", stringValue);
+        } else if (lastSeparator == -1) {
+            return new Date(Long.parseLong(stringValue));
+        } else if (stringValue.length() - lastSeparator - 1 == 2) {
```
Comment (victorg1991) [inline android/library/src/main/java/com/liferay/mobile/screens/ddl/model/DateField.java]: Can you extract this to a function like `isSimpleYearFormat`  or something like that so we have some semantics here? :)

**Comment by mclomello** [comment]
@victorg1991 
Changed two validations of the code where have some data calculated, extracting the calculation for more semantic methods names

**Comment by victorg1991** [comment]
Thanks for the changes :D
This needs a rebase I think, there are a lot of commits that do not belong here

**Comment by victorlaerte** [comment]
@victorg1991 ✅

**Comment by victorg1991** [comment]
Thanks!

**Pull request merged**